protocols/virtualdefund/virtualdefund.go

```diff
@@ -136,7 +171,79 @@ func (o *Objective) clone() Objective {
 
 // Crank inspects the extended state and declares a list of Effects to be executed
 func (o Objective) Crank(secretKey *[]byte) (protocols.Objective, protocols.SideEffects, protocols.WaitingFor, error) {
-	return &o, protocols.SideEffects{}, WaitingForCompleteFinal, errors.New("TODO: UNIMPLEMENTED")
+	updated := o.clone()
```

Comment:
Using a value receiver means the receiver will be a shallow clone of the original. However since slices are reference types the shallow clone of `o` would contain slices that point to the same underlying elements. This means modifying a slice in `Crank` would modify the slice for all copies of `o`.